### PR TITLE
Button doc review

### PR DESCRIPTION
**Pull request by jumeyermattosarruda**
#### What is the purpose of this pull request?
(*This is my first time using .mdx, so be kind to me* 😅) 

To give more context about our documentation:

I wanted to add the FAQ for choosing between components that Davis and I created. 

When we were planning the FAQ structure, we realized that some recommendations are actually repeated for multiple components. For example, the documentation for buttons, links and menus have the same recommendations that apply to all of them. 

Ideally, we thought of storing this text in a separate `.json` file, and embedding it as a snippet on the component files, so we can have it only in one place, instead of adding, maintaining and updating the same text in multiple components. Here you can check out the [slack discussion](https://vtex.slack.com/archives/C02NJHMAY10/p1648064308496109) about this topic.

Since we weren't sure how to deploy that in Admin UI, I started out just adding the documentation for Buttons. This way we could test how they would look like in Admin UI, before moving on to the "snippet in separate file" strategy. This is the PR with my attempt. 

I wasn't so sure about how to add that on `.mdx`, so my train of thought can be seen through my commits' descriptions. I didn't add any `.jsx` sections to my `<BestPractice>` since we created a text-only documentation for the FAQ, but I'm not sure how it will be rendered in Admin UI in the format that I added it. Need some help making the required corrections.

#### What problem is this solving?
Improving design documentation for the button component as a test. 

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Quality improvement (tests or refactors)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Trivial change (small fix or feature that doesn't impact functionalities)
- [x] Requires change to documentation, which has been updated accordingly

#### How does this PR make you feel? 
[:link:](https://giphy.com/gifs/justin-hope-hoping-fingers-crossed-l0NwNrl4BtDD7JCx2)

### CODE REVIEW

**Comment by jumeyermattosarruda** [comment]
Clearly I changed something in the file's structure that made our vercel deployment fail 😮‍💨. Can someone point out what happened so I can learn from this mistake 🙏?

**Comment by lucasaarcoverde** [comment]
> Clearly I changed something in the file's structure that made our vercel deployment fail 😮‍💨. Can someone point out what happened so I can learn from this mistake 🙏?

What went wrong is that you've added the content like that:

```jsx
<Practice
// your content
>

</Practice>
```
This breaks the `<Practice>` component

What you should do to fix:

```jsx
<PageSection
  id={tabToc.designReference[2].id}
  title={tabToc.designReference[2].value}
  explanation="Follow the set of recommendations below to make the best decisions when designing sleek, clear pages."
/>

// your content

</PageSection>
```

PS:
This MDX:
```mdx
## Recommendations
Follow the set of recommendations below to make the best decisions when designing sleek, clear pages.
```
Turns into:
```jsx
<PageSection
  id={tabToc.designReference[2].id}
  title={tabToc.designReference[2].value}
  explanation="Follow the set of recommendations below to make the best decisions when designing sleek, clear pages."
/>
```

**Comment by jumeyermattosarruda** [comment]
@lucasaarcoverde reviewed and followed your instructions! Will take that into consideration for the following components! 

Just not sure about the `/> ` in line 388. The other `<PageSection>` objects don't use the `/` to close the bracket, but your commented recommendation included it. Should I add it, or not?

**Comment by jumeyermattosarruda** [comment]
Also is there a way to preview/stage my branch locally, so I can take a look on how it's rendering, before the PR?

**Comment by lucasaarcoverde** [comment]
> Also is there a way to preview/stage my branch locally, so I can take a look on how it's rendering, before the PR?

Yes, if you are using a terminal to add your changes you can run the following commands:
1. Install project dependencies and build the packages, you just need to run this only once when you create your git branch code.
```bash
yarn && yarn build
```
2. Run the documentation site locally in the following URL: `http://localhost:3000`
```
yarn docs
```

File: packages/admin-ui-docs/docs/guidelines/components/button.mdx [hunk 1]
```diff
@@ -373,6 +379,129 @@ one primary button per page/modal.
...
+</PageSection>
+
+<PageSection
+  id={tabToc.designReference[2].id}
+  title={tabToc.designReference[2].value}
+  explanation="Follow the set of recommendations below to make the best decisions when designing sleek, clear pages."
+>
+
+
+### Components
```
Comment (lucasaarcoverde) [inline packages/admin-ui-docs/docs/guidelines/components/button.mdx:391]:
```suggestion
<PageSection
  id={tabToc.designReference[3].id}
  title={tabToc.designReference[3].value}
>

// Components section content

</PageSection>
```

**Review by lucasaarcoverde — COMMENTED** [review]
Hey Ju, I've just added a suggestion to the way of creating the sections. Using this component `<PageSection>` instead of the raw markdown section it's better because the component reuses the design made by Ale. 

I did a suggestion for the `Components section` but you can do the same to `Variants`, `Position`, `Behavior`, and `Content`

**Comment by lucasaarcoverde** [comment]
Another thing is that the bold words and also the list are not looking good on the site, but you can talk with ale so he can design and define a better style for it. After he finishes defining this we can create a task to improve it from the engineering side, it's not critical

Captura de Tela 2022-05-12 às 13 57 47

**Comment by jumeyermattosarruda** [comment]
@lucasaarcoverde changes were made! I agree that the structure you suggested is much better. We're ready to merge!

**Review by lucasaarcoverde — APPROVED** [review]
Great job Ju! ✨

**Pull request merged**